### PR TITLE
Refactor of job logs

### PR DESCRIPTION
**Pull request by daniel-de-vera**
This PR:

First fixes a panic in JRG list (when a cluster is disconnected):

```bash
$ signadot jrg list
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xc7965f]

goroutine 1 [running]:
github.com/signadot/cli/internal/command/jobrunnergroup.readiness(...)
	/home/ddv/git/signadot/cli/internal/command/jobrunnergroup/printers.go:60
github.com/signadot/cli/internal/command/jobrunnergroup.printRunnerGroupTable(0xc00022d0e0?, {0x1f6cba0?, 0xc00009c028?}, {0xc00022c040, 0x3, 0x0?})
	/home/ddv/git/signadot/cli/internal/command/jobrunnergroup/printers.go:38 +0x27f
github.com/signadot/cli/internal/command/jobrunnergroup.list(0xc00015f988, {0x1f6cba0, 0xc00009c028})
	/home/ddv/git/signadot/cli/internal/command/jobrunnergroup/list.go:40 +0x1b7
github.com/signadot/cli/internal/command/jobrunnergroup.New.newList.func2(0xc0001d9b00?, {0x2e17140?, 0x4?, 0x1cb516d?})
	/home/ddv/git/signadot/cli/internal/command/jobrunnergroup/list.go:22 +0x2c
github.com/spf13/cobra.(*Command).execute(0xc0002d1800, {0x2e17140, 0x0, 0x0})
	/home/ddv/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:916 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004ff200)
	/home/ddv/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:1040 +0x38d
github.com/spf13/cobra.(*Command).Execute(0xc0000061a0?)
	/home/ddv/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:968 +0x13
main.main()
	/home/ddv/git/signadot/cli/cmd/signadot/main.go:11 +0x18
```

Second, in the context of https://github.com/signadot/signadot/pull/4526, I found the SSE library we were using (`github.com/r3labs/sse/v2`) was performing infinite retries in case of errors, what could lead to the duplication of log lines. I tried to find a way to disabled retries, or at least be more selective about them, but couldn't find a simple way to do it.
This is certainly unexpected, for example, what's the point of retrying when getting an HTTP 404 response? Given this was also forcing us to go outside of the SDK, I decided just to find an SSE parsing library, and use it alongside the SDK, having ultimate control over retries in case we want to implement them in the future.

### CODE REVIEW

**Comment by davixcky** [comment]
Minor things i found
1. Is a little behind
2. Error when getting the logs of a cancelled job

https://github.com/signadot/cli/assets/34346877/041e95a5-7b21-468e-986c-29ffe4ddcc48
Screenshot 2024-06-11 at 5 08 51 PM

**Comment by daniel-de-vera** [comment]
> Minor things i found
> 
> 1. Is a little behind

@davixcky, I didn't understand this one, can you expand? Thanks!

**Comment by davixcky** [comment]
> > Minor things i found
> > 
> > 1. Is a little behind
> 
> @davixcky, I didn't understand this one, can you expand? Thanks!

I mean in the video, you can notice how the current implementation is "behind" on logs printing. I'm assuming is cause of the parsing and pipe. Not big deal anyways, but just want it to point out

File: internal/command/logs/command.go [hunk 1]
```diff
@@ -26,71 +27,101 @@ func New(api *config.API) *cobra.Command {
...
+			}()
+
+			go func() {
+				// read the SSE stream
+				_, err := c.JobLogs.StreamJobAttemptLogs(params, nil, writer)
+				writer.Close()
+				errch <- err
+			}()
+
+			return errors.Join(<-errch, <-errch)
```
Comment (scott-cotton) [inline internal/command/logs/command.go:86]: nice idiom!

**Review by davixcky — APPROVED** [review]
LGTM

**Pull request merged**